### PR TITLE
CommandContext: Add a method for getting roomgames

### PR DESCRIPTION
**Pull request by mia-pi-git**
As discussed in #7445.

### CODE REVIEW

File: server/chat.ts [hunk 1]
```diff
@@ -1344,6 +1344,15 @@ export class CommandContext extends MessageContext {
 		}
 		return this.room;
 	}
+	// eslint-disable-next-line @typescript-eslint/type-annotation-spacing
+	getGame<T extends RoomGame>(constructor: new (...args: any[]) => T) {
+		const room = this.requireRoom();
+		if (!room.game) throw new Chat.ErrorMessage(`There is no game going on in this room.`);
+		const game = room.getGame(constructor);
+		// must be a different game
+		if (!game) throw new Chat.ErrorMessage(`There is already a game of ${room.game.title} going on.`);
```
Comment (Zarel) [inline server/chat.ts:1353]: These error messages should probably mention what game _should_ be going on. `constructor.name` would do this, but we could add a static property like `gameName` if we wanted something nicer.

File: server/chat.ts [hunk 2]
```diff
@@ -1344,6 +1344,15 @@ export class CommandContext extends MessageContext {
 		}
 		return this.room;
 	}
+	// eslint-disable-next-line @typescript-eslint/type-annotation-spacing
+	getGame<T extends RoomGame>(constructor: new (...args: any[]) => T) {
```
Comment (Zarel) [inline server/chat.ts:1348]: I would call this `requireGame` instead of `getGame`.

File: server/chat-plugins/blackjack.ts [hunk 3]
```diff
@@ -20,8 +20,6 @@ type SymbolName = 'Hearts' | 'Diamonds' | 'Clubs' | 'Spades';
 export class Blackjack extends Rooms.RoomGame {
 	room: Room;
 
-	blackjack: boolean;
-
 	roomID: RoomID;
```
Comment (Zarel) [inline server/chat-plugins/blackjack.ts:23]: Are these properties necessary? They should already be defined in `RoomGame`, right?

Comment (mia-pi-git) [reply server/chat-plugins/blackjack.ts:23]: In the name of checking later: my response that i sent on discord (roughly)

Possibly? I don't maintain blackjack, so i didn't want to delete anything that wasn't directly relevant to my change.

File: server/chat.ts (hunk 1 above)
Comment (Zarel) [reply server/chat.ts:1353]:
```suggestion
		if (!room.game) throw new Chat.ErrorMessage(`This command requires a game of ${constructor.name} (this room has no game).`);
		const game = room.getGame(constructor);
		// must be a different game
		if (!game) throw new Chat.ErrorMessage(`This command requires a game of ${constructor.name} (this game is ${room.game.title}).`);
```

File: server/chat.ts [hunk 4]
```diff
@@ -1347,10 +1347,10 @@ export class CommandContext extends MessageContext {
 	// eslint-disable-next-line @typescript-eslint/type-annotation-spacing
 	requireGame<T extends RoomGame>(constructor: new (...args: any[]) => T) {
 		const room = this.requireRoom();
-		if (!room.game) throw new Chat.ErrorMessage(`There is no game going on in this room.`);
+		if (!room.game) throw new Chat.ErrorMessage(`This command requires a game of ${constructor.name} (this room has no game).`);
```
Comment (Zarel) [inline server/chat.ts:1350]:
```suggestion
		if (!room.game) {
			throw new Chat.ErrorMessage(`This command requires a game of ${constructor.name} (this room has no game).`);
		}
```

File: server/chat.ts [hunk 5]
```diff
@@ -1347,10 +1347,10 @@ export class CommandContext extends MessageContext {
...
 		const game = room.getGame(constructor);
 		// must be a different game
-		if (!game) throw new Chat.ErrorMessage(`There is already a game of ${room.game.title} going on.`);
+		if (!game) throw new Chat.ErrorMessage(`This command requires a game of ${constructor.name} (this game is ${room.game.title}).`);
```
Comment (Zarel) [inline server/chat.ts:1353]:
```suggestion
		if (!game) {
			throw new Chat.ErrorMessage(`This command requires a game of ${constructor.name} (this game is ${room.game.title}).`);
		}
```

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**